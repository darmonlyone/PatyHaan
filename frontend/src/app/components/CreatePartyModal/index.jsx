import React from "react";
import { mdiArrowLeft } from "@mdi/js";

import InputField from "app/common/InputField";
import LinkButton from "app/common/LinkButton";

import useCreatePartyHandle from "app/hooks/useCreatePartyHandle"

const CreatePartyModal = ({ onCloseHandle }) => {
	const {
		partyName,
		amount,
		imageUrl,
		onPartyNameChange,
		onAmountChange,
		onImageUrlChange,
		onCreatePartySubmit,
	} = useCreatePartyHandle(onCloseHandle)

	return (
		<div className="fixed z-50 flex justify-center items-center h-screen w-full bg-gray-200 bg-opacity-60">
			<div className="bg-white h-full sm:h-auto w-full sm:w-auto sm:rounded-lg">
				<div className="w-full bg-white h-12 shadow-md sm:shadow-none flex items-center relative sm:rounded-t-md sm:mt-2">
					<svg
						className="h-10 w-10 sm:h-8 sm:w-8 absolute left-1 top-1"
						viewBox="0 0 24 24"
						onClick={onCloseHandle}
					>
						<path d={mdiArrowLeft} />
					</svg>
					<span className="text-lg font-bold text-center w-full">
						สร้างปาร์ตี้
					</span>
				</div>
				<div className="w-full h-full flex flex-col justify-center items-center px-10 mb-8">
					<InputField
						testid="partyname"
						title="ชื่อปาร์ตี้"
						value={partyName}
						onChange={onPartyNameChange}
					/>
					<InputField
						testid="amount"
						title="จำนวนคนที่ขาด"
						type="number"
						value={amount}
						onChange={onAmountChange}
					/>
					<InputField
						testid="imageurl"
						title="ลิงค์รุปภาพ (กรณีมีรูปภาพ)"
						value={imageUrl}
						onChange={onImageUrlChange}
					/>
					<LinkButton
						testid="submitbutton"
						onClick={onCreatePartySubmit}
						className="w-full bg-blue-400 hover:bg-blue-500 mt-4"
					>
						สร้างปาร์ตี้
					</LinkButton>
				</div>
			</div>
		</div>
	);
};

export default CreatePartyModal;
