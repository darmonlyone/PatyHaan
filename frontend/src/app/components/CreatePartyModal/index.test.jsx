import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";

import CreatePartyModal from ".";

const onCloseHandle = jest.fn();

jest.mock("app/hooks/useCreatePartyHandle", () => {
	return () => {
		return {
			partyName: "test",
			amount: 5,
			imageUrl: "test url",
			onPartyNameChange: jest.fn(),
			onAmountChange: jest.fn(),
			onImageUrlChange: jest.fn(),
			onCreatePartySubmit: jest.fn(),
		};
	};
});

it("should render CreatePartyModal with changed state value", async () => {
	const { asFragment, getByTestId, getByText } = render(
		<Router>
			<CreatePartyModal onCloseHandle={onCloseHandle} />
		</Router>
	);

	expect(getByTestId("partyname").value).toEqual("test");
	expect(getByTestId("amount").value).toEqual("5");
	expect(getByTestId("imageurl").value).toEqual("test url");
	expect(getByText("ชื่อปาร์ตี้")).toBeDefined();
	expect(asFragment()).toMatchSnapshot();

});
