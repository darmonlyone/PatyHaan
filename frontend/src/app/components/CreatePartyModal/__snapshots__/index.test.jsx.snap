// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CreatePartyModal with changed state value 1`] = `
<DocumentFragment>
  <div
    class="fixed z-50 flex justify-center items-center h-screen w-full bg-gray-200 bg-opacity-60"
  >
    <div
      class="bg-white h-full sm:h-auto w-full sm:w-auto sm:rounded-lg"
    >
      <div
        class="w-full bg-white h-12 shadow-md sm:shadow-none flex items-center relative sm:rounded-t-md sm:mt-2"
      >
        <svg
          class="h-10 w-10 sm:h-8 sm:w-8 absolute left-1 top-1"
          viewBox="0 0 24 24"
        >
          <path
            d="M20,11V13H8L13.5,18.5L12.08,19.92L4.16,12L12.08,4.08L13.5,5.5L8,11H20Z"
          />
        </svg>
        <span
          class="text-lg font-bold text-center w-full"
        >
          สร้างปาร์ตี้
        </span>
      </div>
      <div
        class="w-full h-full flex flex-col justify-center items-center px-10 mb-8"
      >
        <div
          class="my-2 w-full"
        >
          <span
            class="mb-2 text-sm"
          >
            ชื่อปาร์ตี้
          </span>
          <input
            class="text-md w-full"
            data-testid="partyname"
            required=""
            type="text"
            value="test"
          />
        </div>
        <div
          class="my-2 w-full"
        >
          <span
            class="mb-2 text-sm"
          >
            จำนวนคนที่ขาด
          </span>
          <input
            class="text-md w-full"
            data-testid="amount"
            required=""
            type="number"
            value="5"
          />
        </div>
        <div
          class="my-2 w-full"
        >
          <span
            class="mb-2 text-sm"
          >
            ลิงค์รุปภาพ (กรณีมีรูปภาพ)
          </span>
          <input
            class="text-md w-full"
            data-testid="imageurl"
            required=""
            type="text"
            value="test url"
          />
        </div>
        <button
          class="w-full bg-blue-400 hover:bg-blue-500 mt-4"
        >
          สร้างปาร์ตี้
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
